PC/winreg.c

```diff
@@ -1319,6 +1320,78 @@ winreg_FlushKey_impl(PyObject *module, HKEY key)
     Py_RETURN_NONE;
 }
 
+/*[clinic input]
+winreg.GetValue
+
+    key: HKEY
+        An already open key, or any one of the predefined HKEY_* constants.
+    sub_key: Py_UNICODE(accept={str, NoneType})
+        A string that names the subkey with which the value is associated.
+    name: Py_UNICODE(accept={str, NoneType})
+        A string indicating the value to query.
+    /
+
+Retrieves the type and data for the specified registry value.
+
+Behaves mostly like QueryValueEx(), but you needn't OpenKey() and CloseKey() if the key is any one of the predefined HKEY_* constants.
+
+The return value is a tuple of the value and the type_id.
+[clinic start generated code]*/
+
+static PyObject *
+winreg_GetValue_impl(PyObject *module, HKEY key, const Py_UNICODE *sub_key,
+                     const Py_UNICODE *name)
+/*[clinic end generated code: output=b8bb6c5c588205e1 input=c2e4b5b8f973c595]*/
+{
+    long rc;
+    BYTE *retBuf, *tmp;
+    DWORD bufSize = 0, retSize;
+    DWORD typ;
+    PyObject *obData;
+    PyObject *result;
+
+    if (PySys_Audit("winreg.GetValue", "nuu",
+                    (Py_ssize_t)key, sub_key, name) < 0) {
+        return NULL;
+    }
+
+    rc = RegGetValueW(key, sub_key, name, 0xffff, NULL, NULL, &bufSize);
```

Comment:
This should use the macro `RRF_RT_ANY` (0xFFFF) for the flags. I suggest making it the default value and supporting the flags as an argument, with at least `RRF_NOEXPAND` defined as a constant in winreg. I suggest also manually defining `RRF_SUBKEY_WOW6464KEY` (0x00010000) and `RRF_SUBKEY_WOW6432KEY` (0x00020000) macros and winreg constants, which require Window 10.  Python 3.11 still supports Windows 8.1, but Windows 10 is used predominantly.

Reply:
Would supporting flags as an argument be complicated?
I see all the RRF_RT_xxx and RRF_NOEXPAND and RRF_SUBKEY_WOW6464KEY are not compatible with existed winreg.KEY_WOW64_64KEY and winreg.REG_DWORD etc.

Reply:
By complicated, do you mean confusing? It's a matter of providing winreg constants and documenting them (e.g. something simple like "can be one or more of the `winreg.RRF_*` constants"). Except there's no point in explicitly supporting `RRF_ZEROONFAILURE`. If new flags are added in the future, they can be passed in manually, i.e. I don't think Python should validate the flags argument.